src/ProgressLogging.jl

```diff
@@ -499,4 +485,25 @@ _comprehension(iter_exprs, body) =
     Expr(:comprehension, Expr(:generator, body, iter_exprs...))
 _for(iter_exprs, body) = Expr(:for, Expr(:block, reverse(iter_exprs)...), body)
 
+taccumulate(op, ::Tuple{}) = ()
+function taccumulate(op::F, xs::Tuple) where {F}
+    ys, = foldl(Base.tail(xs); init=((xs[1],), xs[1])) do (ys, acc), x
+        acc = op(acc, x)
+        (ys..., acc), acc
+    end
+    return ys
+end
+
+function make_count_to_frac(iterators...)
```

Comment:
This is factored out as a function (rather than inline code) so that manual `@gensym` is slightly easier.  Also, I think this may generate faster code (not tested).